cob_image_flip/ros/src/image_flip.cpp

```diff
@@ -429,14 +496,14 @@ void ImageFlip::pcCallback(const sensor_msgs::PointCloud2::ConstPtr& point_cloud
 		rot_mat = cv::getRotationMatrix2D(center, -rotation_angle, 1.0);
 		if (switch_aspect_ratio==true)
 		{
-			rot_mat.at<double>(0,2) += 0.5*((double)point_cloud_out_msg->width - (double)point_cloud_msg->width);
-			rot_mat.at<double>(1,2) += 0.5*((double)point_cloud_out_msg->height - (double)point_cloud_msg->height);
-		}
+		    atForDiffImTypes(rot_mat, 0, 2, 0.5*((double)point_cloud_out_msg->width - (double)point_cloud_msg->width), false);
+		    atForDiffImTypes(rot_mat, 1, 2, 0.5*((double)point_cloud_out_msg->height - (double)point_cloud_msg->height), false);
+        }
 		cv::Mat rot_mat_inv = rot_mat.clone();
-		rot_mat_inv.at<double>(0,1) = rot_mat.at<double>(1,0);
```

Comment:
Just leave it the old way in this case, because the rotation matrix is always the same format CV_64FC1.